### PR TITLE
Master multicompany stock whe

### PR DESCRIPTION
**Pull request by Whenrow**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_location.py [hunk 1]
```diff
@@ -174,6 +174,11 @@ def should_bypass_reservation(self):
         self.ensure_one()
         return self.usage in ('supplier', 'customer', 'inventory', 'production') or self.scrap_location
 
+    @api.constrains('location_id', 'location_id')
```
Comment (mreficent) [inline addons/stock/models/stock_location.py]: `@api.constrains('company_id', 'location_id')`, right? 🤔

Comment (Whenrow) [reply addons/stock/models/stock_location.py]: @mreficent oups, you're right ! thanks ;)

File: addons/stock/views/stock_picking_views.xml [hunk 2]
```diff
@@ -262,6 +265,7 @@
                             <field name="picking_type_id" attrs="{'readonly': [('state', '!=', 'draft')]}"/>
                             <field name="location_id" options="{'no_create': True}" groups="stock.group_stock_multi_locations" attrs="{'invisible': [('picking_type_code', '=', 'incoming')]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
                             <field name="location_dest_id" options="{'no_create': True}" groups="stock.group_stock_multi_locations" attrs="{'invisible': [('picking_type_code', '=', 'outgoing')]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
+                            <field name="picking_type_id" attrs="{'readonly': [('state', '!=', 'draft')]}"/>
```
Comment (sle-odoo) [inline addons/stock/views/stock_picking_views.xml]: why?

File: addons/stock/models/stock_location.py [hunk 3]
```diff
@@ -174,6 +174,11 @@ def should_bypass_reservation(self):
         self.ensure_one()
         return self.usage in ('supplier', 'customer', 'inventory', 'production') or self.scrap_location
 
+    @api.constrains('company_id', 'location_id')
+    def _check_company(self):
+        if self.company_id and self.location_id.company_id and self.company_id != self.location_id.company_id:
+            raise UserError(_('The company cannot be different than the parent''s one'))
```
Comment (mreficent) [inline addons/stock/models/stock_location.py]:
```suggestion
            raise UserError(_('The company cannot be different than the parent\'s one'))
```
🤔

**Pull request closed without merging**